streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -217,14 +217,18 @@ private long updateOffsetLimits(final TopicPartition partition) {
             throw new IllegalArgumentException("Topic is not both a source and a changelog: " + partition);
         }
 
-        updateableOffsetLimits.remove(partition);
+        final Map<TopicPartition, Long> newLimits = committedOffsetForPartition(updateableOffsetLimits);
+        final Long previousLimit = offsetLimits.get(partition);
+        final Long newLimit = newLimits.get(partition);
 
-        final long newLimit = committedOffsetForPartition(partition);
-        final long previousLimit = offsetLimits.put(partition, newLimit);
-        if (previousLimit > newLimit) {
+        if (previousLimit != null && previousLimit > newLimit) {
```

Comment:
We should either do this check for all updated limits or none.

Reply:
Ack. Thinking about this a bit more, I think we do not need the `updateableOffsetLimits` since it always have the same topic-partitions with `offsetLimits`, and instead we just keep a flag and always ask for all topic-partitions in `offsetLimits`. WDYT?

Reply:
+1 now that we're doing the call in batch.